src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,298 @@
+    protected JedisPoolConfig redisConnectionPoolConfig = null;
+
+    //Database info
+    private String dbHost = "localhost";
+    private int dbPort = 2222;
+    private int dbTimeout = 0;
+    private String dbPassword = "";
+
+    private static final String ISO_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ssZ";
+
+
+    /**
+     * The lifecycle event support for this component.
+     */
+    protected LifecycleSupport lifecycle = new LifecycleSupport(this);
+
+    @Override
+    public void load() throws ClassNotFoundException, IOException {
+    }
+
+    @Override
+    public void unload() throws IOException {
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START Sessions
+    //------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public int getRejectedSessions() {
+        // Essentially do nothing.
+        return 0;
+    }
+
+    @Override
+    public Session createSession(String requestedSessionId) {
+
+        Session session = createEmptySession();
+
+        // Initialize the properties of the new session and return it
+        session.setId(UUID.randomUUID().toString());
+        session.setNew(true);
+        session.setValid(true);
+        session.setCreationTime(System.currentTimeMillis());
+        session.setMaxInactiveInterval(this.maxInactiveInterval);
+
+        try {
+            Jedis jedis = redisConnectionPool.getResource();
+            saveToRedis(jedis, session);
+            redisConnectionPool.returnResourceObject(jedis);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return session;
+    }
+
+    /**
+     * Add session to this manager
+     * @param session
+     */
+    @Override
+    public void add(Session session) {
+        try {
+            Jedis jedis = redisConnectionPool.getResource();
+            saveToRedis(jedis, session);
+            redisConnectionPool.returnResourceObject(jedis);
+        } catch (IOException ex) {
+            log.warn("Unable to add to session manager store: " + ex.getMessage());
+            throw new RuntimeException("Unable to add to session manager store.", ex);
+        }
+    }
+
+    @Override
+    public Session findSession(String id) throws IOException {
+        Session session = null;
+
+        //Get connection to redis
+        Jedis jedis = redisConnectionPool.getResource();
+
+        Hashtable<String, Object> metadata =
+                (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(jedis.get((id + REDIS_METADATA_KEY).getBytes())));
+
+        Hashtable<String, Object> attributes =
+                (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(jedis.get((id + REDIS_ATTRIBUTES_KEY).getBytes())));
+
+        session = getSession(metadata, attributes);
+
+        redisConnectionPool.returnResourceObject(jedis);
```

Comment:
You will never return this resource object if there is an exception. You should define a method that takes a `Function<Jedis,T>` and performs all the connection pool handling. Pass a lambda into that function.